src/test/java/wolox/training/controllers/BookControllerTest.java

```diff
@@ -0,0 +1,185 @@
+
+        given(bookRepository.findFirstByAuthor("Paolo")).willReturn(java.util.Optional.of(bookTest));
+
+        // When + Then
+        mvc.perform(MockMvcRequestBuilders.get("/api/books?author=Francisco")
+                .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isNotFound());
+        System.out.println("whenGetABookByAuthorNonExist Passed");
+    }
+
+
+    @Test
+    void whenGetABookById_thenReturnJson() throws Exception {
+        //Given
+        Book bookTest = new Book(1L,"Mystery","Paolo", "image.png", "Sorcerer's Stone", "-", "Salamander","2000", 250,"123456789");
+
+        given(bookRepository.findById(bookTest.getId())).willReturn(java.util.Optional.of(bookTest));
+
+        // When + Then
+        mvc.perform(MockMvcRequestBuilders.get("/api/books/1")
+                .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$").exists())
+                .andExpect(jsonPath("$", isA(LinkedHashMap.class)))
+                .andExpect(jsonPath("$.*").exists())
+                .andExpect(jsonPath("$.*", hasSize(is(10))))
+                .andExpect(jsonPath("$.id", is(bookTest.getId().intValue())))
+                .andExpect(jsonPath("$.image", is(bookTest.getImage())));
+        System.out.println("whenGetABookById Passed");
+    }
+
+    @Test
+    void whenAddABook_thenReturnOk() throws Exception {
+        //Given
+        Book bookTest = new Book(1L,"Mystery","Paolo", "image.png", "Sorcerer's Stone", "-", "Salamander","2000", 250,"123456789");
+
+        given(bookRepository.save(bookTest)).willReturn(bookTest);
+
+        String json = mapper.writeValueAsString(bookTest);
+
+        // When + Then
+        mvc.perform(MockMvcRequestBuilders.post("/api/books")
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(json)
+                .accept(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk());
+        System.out.println("whenAddABook Passed");
+    }
+
+
+    @Test
+    void whenUpdateABook_thenReturnOk() throws Exception {
+        //Given
+        Book bookTest = new Book(1L,"Mystery","Paolo", "image.png", "Sorcerer's Stone", "-", "Salamander","2000", 250,"123456789");
+
+        given(bookRepository.save(bookTest)).willReturn(bookTest);
+
+        bookTest.setTitle("Chamber of Secrets");
+        given(bookRepository.save(bookTest)).willReturn(bookTest);
+        given(bookRepository.findById(bookTest.getId())).willReturn(java.util.Optional.of(bookTest));
+
+        String json = mapper.writeValueAsString(bookTest);
+
+        // When + Then
+
+        mvc.perform(MockMvcRequestBuilders.put("/api/books/1")
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(json)
+                .accept(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk());
+        System.out.println("whenUpdateABook Passed");
```

Comment:
This print is not necessary, as it is a test, the result is already shown by the console